src/main/scala/akka/persistence/jdbc/query/JournalSequenceActor.scala

```diff
@@ -139,15 +152,10 @@ class JournalSequenceActor(readJournalDao: ReadJournalDao, config: JournalSequen
           (newMax, currentElement, newMissing)
       }
 
-    val newMissingByCounter =
-      (missingByCounter + (moduloCounter -> missingElems))
-        .map {
-          case (key, value) =>
-            key -> value.filter(missingId => missingId > nextMax)
-        }
+    val newMissingByCounter = missingByCounter + (moduloCounter -> missingElems)
 
     // did we detect gaps in the current batch?
-    val noGapsFound = newMissingByCounter.values.forall(_.isEmpty)
+    val noGapsFound = missingElems.isEmpty
```

Comment:
Since we no longer filter elements which were already missing in previous iterations. I was able to make this much more simple